MiniSim/Menu.swift

```diff
@@ -121,103 +121,125 @@ class Menu: NSMenu {
-            
-            let iosDevicesCount = self.devices.filter({ $0.platform == .ios }).count
-            self.safeInsertItem(menuItem, at: isAndroid && UserDefaults.standard.enableiOSSimulators ? (isFirst ? index : iosDevicesCount) + 3 : 1)
+        let platformSections: [DeviceListSection] = sections
+        for section in platformSections {
+            let devices = filter(devices: sortedDevices, for: section)
+            let menuItems = devices.map { createMenuItem(for: $0) }
+            self.updateSection(with: Array(menuItems), section: section)
         }
     }
     
-    private func populateAndroidSubMenu(booted: Bool) -> NSMenu {
-        let subMenu = NSMenu()
-        for item in AndroidSubMenuItem.allCases {
-            if item == AndroidSubMenuItem.customCommand {
-                continue
-            }
-            
-            let menuItem = item.menuItem
-            menuItem.target = self
-            menuItem.action = #selector(androidSubMenuClick)
-            if item.needBootedDevice && !booted {
-                continue
-            }
-            if item.bootsDevice && booted {
-                continue
-            }
-            subMenu.addItem(menuItem)
+    var sections: [DeviceListSection] {
+        var sections: [DeviceListSection] = []
+        if UserDefaults.standard.enableAndroidEmulators {
+            sections.append(.android)
         }
-        
-        for item in DeviceService.getCustomCommands(platform: .android) {
-            let menuItem = AndroidSubMenuItem.customCommand.menuItem
-            menuItem.target = self
-            menuItem.action = #selector(androidSubMenuClick)
-            if item.needBootedDevice && !booted {
-                continue
-            }
-            if item.bootsDevice ?? false && booted {
-                continue
-            }
-            menuItem.image = NSImage(systemSymbolName: item.icon, accessibilityDescription: item.name)
-            menuItem.title = item.name
-            subMenu.addItem(menuItem)
+        if UserDefaults.standard.enableiOSSimulators {
+            sections.append(.iOS)
         }
-        return subMenu
+        return sections
     }
     
-    private func populateIOSSubMenu(booted: Bool) -> NSMenu {
-        let subMenu = NSMenu()
-        for item in IOSSubMenuItem.allCases {
-            if item == IOSSubMenuItem.customCommand {
-                continue
+    private func filter(devices: [Device], for section: DeviceListSection) -> [Device] {
+        let platform: Platform = section == .iOS ? .ios : .android
+        return devices.filter { $0.platform == platform }
+    }
+    
+    private func updateSection(with items: [NSMenuItem], section: DeviceListSection) {
+        guard let header = self.items.first(where: { $0.tag == section.rawValue }),
+              let startIndex = self.items.firstIndex(of: header) else {
+            return
+        }
+        
+        var count = 0
+        items.forEach { menuItem in
```

Comment:
Can we use `.enumerated()` here instead of count? I think it's more elegant solution

Reply:
Yeah, agree. Looks better, thanks for suggestion.